File: albumloader/src/main/java/com/lee/album/router/GalleryEngine.java
package com.lee.album.router;

import android.app.Activity;
import android.content.Context;

import androidx.fragment.app.Fragment;

import java.lang.ref.WeakReference;

/**
 * Author : 李嘉伦
 * e-mail : lijialun@angogo.cn
 * date   : 2022/1/1014:07
 * Package: com.lee.album.router
 * desc   :
 */
public class GalleryEngine {
    private WeakReference<Activity> mWeakActivity;
    private WeakReference<Fragment> mWeakFragment;

    private GalleryEngine(Activity activity) {
        this(activity, null);
    }

    private GalleryEngine(Fragment fragment) {
        this(fragment.getActivity(), fragment);
    }

    private GalleryEngine(Activity activity, Fragment fragment) {
        mWeakActivity = new WeakReference<>(activity);
        mWeakFragment = new WeakReference<>(fragment);
    }

    public static GalleryEngine from(Activity activity) {
        return new GalleryEngine(activity);
    }

    public static GalleryEngine from(Fragment fragment) {
        return new GalleryEngine(fragment);
    }


    public GalleryBuilder setGalleryBuilder(Context context) {
        return new GalleryBuilder(context,this);
    }

    public Activity getActivity() {
        return mWeakActivity.get();
    }

    public Fragment getFragment() {
        return mWeakFragment.get();
    }

} 

File: albumloader/src/main/java/com/lee/album/router/GalleryEngine.kt
package com.lee.album.router

import android.app.Activity
import android.content.Context
import androidx.fragment.app.Fragment
import com.lee.album.router.GalleryBuilder
import com.lee.album.router.GalleryEngine
import java.lang.ref.WeakReference

/**
 * Author : 李嘉伦
 * e-mail : lijialun@angogo.cn
 * date   : 2022/1/1014:07
 * Package: com.lee.album.router
 * desc   :
 */
class GalleryEngine private constructor(activity: Activity?, fragment: Fragment? = null) {
    private val mWeakActivity: WeakReference<Activity?> = WeakReference(activity)
    private val mWeakFragment: WeakReference<Fragment?> = WeakReference(fragment)

    private constructor(fragment: Fragment) : this(fragment.activity, fragment) {}

    fun setGalleryBuilder(context: Context?): GalleryBuilder {
        return GalleryBuilder(context!!, this)
    }

    val activity: Activity?
        get() = mWeakActivity.get()
    val fragment: Fragment?
        get() = mWeakFragment.get()

    companion object {
        fun from(activity: Activity?): GalleryEngine {
            return GalleryEngine(activity)
        }

        fun from(fragment: Fragment): GalleryEngine {
            return GalleryEngine(fragment)
        }
    }

}